CMake/CMakeLists_legacy.cmake.in

```diff
@@ -117,6 +117,7 @@ set(FlatBuffers_Library_SRCS
   include/flatbuffers/flatbuffer_builder.h
   include/flatbuffers/flatbuffers.h
   include/flatbuffers/flexbuffers.h
+  include/flatbuffers/flex_flat_util.h
```

Comment:
@ilya-lavrenov

Reply:
I can do the change, but would really prefer people to use a more modern cmake version. I'm tempted to remove this legacy file.

Reply:
Currently, it's kind of dead code and since our project has a requirement cmake 3.13, we faced with issue issue once we added the flatbuffers as submodule. I suppose the options are:
- Completely remove legacy cmake scripts, because users cannot compile `flatc` because of missed files, incorrect `POST_BUILD` steps.
- Fix legacy cmake since it's still a part of repository.

Maybe one more option is to generalize main `CMakeLists.txt` to support older cmake? Actually, cmake 3.10 or 3.13 is powerful enough to cover all use-cases of `flatbuffers`. I can try to do it if you are OK with the approach.

Reply:
> our project has a requirement cmake 3.13

That's a minimum version right? Or you only support up to 3.13? 

> Currently, it's kind of dead cod

Yep, I should have removed it a while ago.

> Completely remove legacy cmake scripts, because users cannot compile flatc because of missed files, incorrect POST_BUILD steps.

I would prefer this. Less maintenance on our end. I believe 3.16 was the minimum version to support all the features I used, but I forget. If you could make the main `CMakeLists.txt` 3.13 compatible, I would be more inclined for that.

Reply:
> That's a minimum version right? Or you only support up to 3.13?

Yes, 3.13 is a minimum required version. 

> If you could make the main CMakeLists.txt 3.13 compatible, I would be more inclined for that

Great, let me try to do it.

Reply:
> Yes, 3.13 is a minimum required version.

Ok, what prevents you from personally running a newer cmake then, or if you include flatbuffers, requiring the 3.16 as the new minimum? 

3.16 was released 2019-11-26
3.13 was release 2018-11-20

Both are 4+ years old by now, and between the two only a year separates them.

Reply:
> Ok, what prevents you from personally running a newer cmake then

I can install newer version on my machine, but we have a lot of users who does not have 3.16 on their machines and we don't want  users to install newer versions for cases when we can try to avoid it.

E.g. a lot of PRC users still have CentOS 7 where `cmake3` executable has version 3.14.6.